src/crypto/OlmDevice.ts

```diff
@@ -1222,7 +1222,9 @@ export class OlmDevice {
                     if (session === null || sessionData === null) {
                         if (withheld) {
                             error = new DecryptionError(
-                                DecryptionFailureCode.MEGOLM_UNKNOWN_INBOUND_SESSION_ID,
+                                withheld.code === "m.unverified"
```

Comment:
I think it would be clearer if the code was calculated outside the `DecryptionError(...)`, as in:

```
const failureCode = withheld.code === "m.unverified" ? ...
error = new DecryptionError(
  failureCode,
  ...
)
```

(and similarly below), but YMMV